test/runtests.jl

```diff
@@ -168,11 +216,11 @@ end
 end
 
 # detect incorrect numbers of fields
-@test_throws AssertionError @match Foo(x) = Foo(1,2)
-@test_throws AssertionError @match Foo(x,y,z) = Foo(1,2)
+@test_throws ErrorException @match Foo(x) = Foo(1,2)
+@test_throws ErrorException @match Foo(x,y,z) = Foo(1,2)
 
 # ...even if the pattern is not reached
-@test_throws AssertionError (@match Foo(1,2) begin
+@test_throws ErrorException (@match Foo(1,2) begin
```

Comment:
(Why did these exception types need to change? Were the tests failing beforehand?)

Reply:
The code for structs changed from `@assert` to `error()`, so this was needed to keep the test working.

Reply:
Ah i see that now. thanks!